public/app/features/dashboard/dashgrid/DashboardPanel.tsx

```diff
@@ -1,24 +1,23 @@
 // Libraries
 import React, { PureComponent } from 'react';
 import classNames from 'classnames';
-
 // Utils & Services
 import { getAngularLoader, AngularComponent } from '@grafana/runtime';
 import { importPanelPlugin } from 'app/features/plugins/plugin_loader';
-
 // Components
 import { AddPanelWidget } from '../components/AddPanelWidget';
 import { DashboardRow } from '../components/DashboardRow';
 import { PanelChrome } from './PanelChrome';
 import { PanelEditor } from '../panel_editor/PanelEditor';
 import { PanelResizer } from './PanelResizer';
-
 // Types
 import { PanelModel, DashboardModel } from '../state';
 import { PanelPluginMeta, PanelPlugin } from '@grafana/ui/src/types/panel';
 import { AutoSizer } from 'react-virtualized';
+import { mouseMoveEvent } from '@grafana/data';
+import { EventsContextApi, withEvents } from '../../../core/utils/EventsProvider';
 
-export interface Props {
+export interface Props extends EventsContextApi<React.MouseEvent<HTMLDivElement, MouseEvent>, React.MouseEvent<HTMLDivElement, MouseEvent>> {
```

Comment:
I made it so you have to define which EventTypes you subscribe to and which EventTypes you can publish. Might be overkill?